modules/distribution/utils_delegator_rewards.go

```diff
@@ -10,21 +10,23 @@ import (
 
 // refreshDelegatorsRewardsAmounts refreshes the rewards associated with all the delegators for the given height,
 // deleting the ones existing and downloading them from scratch.
-func (m *Module) refreshDelegatorsRewardsAmounts(height int64) {
-	interval := m.cfg.RewardsFrequency
-	if interval == 0 {
-		log.Debug().Str("module", "distribution").Msg("delegator rewards refresh interval set to 0. Skipping refresh")
-		return
-	}
-
-	hasRewards, err := m.db.HasDelegatorRewards()
-	if err != nil {
-		log.Error().Str("module", "distribution").Err(err).Int64("height", height).
-			Msg("error while checking delegators reward")
-	}
-
-	if hasRewards && height%interval != 0 {
-		return
+func (m *Module) refreshDelegatorsRewardsAmounts(height int64, checkInterval bool) {
+	if checkInterval {
```

Comment:
I don't like much this approach of having a boolean field just to tell whether or not to check the interval before updating the amounts. Instead, I think the whole interval check can be safely moved inside the `HandleBlock` method since it's used only there. Then this method can become more easily readable and can be safely called from the messages handlers as well

Reply:
Hi @RiccardoM , sure I just modified, is it like this?